flang/lib/Optimizer/OpenMP/MapInfoFinalization.cpp

```diff
@@ -151,16 +152,23 @@ class MapInfoFinalizationPass
     mlir::Location loc = boxMap->getLoc();
     assert(allocaBlock && "No alloca block found for this top level op");
     builder.setInsertionPointToStart(allocaBlock);
-    auto alloca = builder.create<fir::AllocaOp>(loc, descriptor.getType());
+
+    mlir::Type allocaType = descriptor.getType();
+    if (fir::isTypeWithDescriptor(allocaType) &&
+        !mlir::isa<fir::BaseBoxType>(descriptor.getType()))
```

Comment:
```suggestion
    if (fir::isBoxAddress(allocaType))
```

Reply:
Thank you very much Jean, I'll add that in the next update!